sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -31,10 +31,17 @@ import org.apache.spark.sql.catalyst.util.{CaseInsensitiveMap, CompressionCodecs
  * Most of these map directly to Jackson's internal options, specified in [[JsonParser.Feature]].
  */
 private[sql] class JSONOptions(
-    @transient private val parameters: CaseInsensitiveMap)
+    @transient private val parameters: CaseInsensitiveMap,
+    defaultColumnNameOfCorruptRecord: String)
```

Comment:
Previously the `JSONOptions` instance was always passed around with a `columnNameOfCorruptRecord` value. This just makes it a field in `JSONOptions` instead to put all options in one place. Since it's a required option it made more sense to use a field instead making an entry in the `CaseInsensitiveMap`.